src/screens/CreatePost/elements/useHastagMention.js

```diff
@@ -6,14 +6,14 @@ const useHastagMention = (initial = '') => {
   const [formattedText, setFormattedText] = React.useState(initial);
   const [hashtags, setHashtags] = React.useState([]);
 
-  const handleStateHashtag = (text, setHashtagState = null) => {
-    const hashtagMention = handleHastagMention(text, hashtags, setHashtagState);
+  const handleStateHashtag = (text) => {
+    const hashtagMention = handleHastagMention(text, hashtags);
     setFormattedText(hashtagMention);
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleStateHashtag` function has been simplified by removing the optional `setHashtagState` parameter. This change seems to be an improvement as it reduces complexity and potential side effects from manipulating state outside of this hook.

```diff
-  const handleStateHashtag = (text, setHashtagState = null) => {
-    const hashtagMention = handleHastagMention(text, hashtags, setHashtagState);
+  const handleStateHashtag = (text) => {
+    const hashtagMention = handleHastagMention(text, hashtags);
     setFormattedText(hashtagMention);
   };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->